pkg/agent/datamodel/types.go

```diff
@@ -497,6 +497,7 @@ type KubernetesConfig struct {
 	ProxyMode                         KubeProxyMode     `json:"kubeProxyMode,omitempty"`
 	PrivateAzureRegistryServer        string            `json:"privateAzureRegistryServer,omitempty"`
 	OutboundRuleIdleTimeoutInMinutes  int32             `json:"outboundRuleIdleTimeoutInMinutes,omitempty"`
+	EnableACRTeleportPlugin           *bool             `json:"enableACRTeleportPlugin,omitempty"`
```

Comment:
I'm thinking about the long term goal for agentbaker - ideally it should only get one input from the NodeConfiguration, instead of containerservice or agent pool, as those are AKS or AKS-E concept, which isn't a necessary concept for AgentBaker.  Ideally RP should consolidate all the infor and just provide the goal state in NodeConfiguration. If we follow that principal, it seems we should stop adding more property to this agent pool struct, What do you think?

Reply:
This will also address a recent concern from a discussion around the private cluster - we should decouple cse from RP as much as we can, by moving this outside of the agentpool, we're forcing RP explicitly pass it to cse contract, which is better than implicit dependency on RP and make it easier to decouple in future, if any

Reply:
Good suggestion. I wasn't 100% aware of our new pattern for NodeBootstrapConfigs. Moving model changes there.